client/app/telecom/sms/sms/templates/telecom-sms-sms-templates.controller.js

```diff
@@ -78,12 +83,18 @@ angular.module("managerApp").controller("TelecomSmsSmsTemplatesCtrl", class Tele
     refresh () {
         this.api.sms.templates.resetCache();
         this.api.sms.templates.resetQueryCache();
+        this.templates.paginated = [];
         this.templates.raw = null;
         this.templates.isLoading = true;
         return this.api.sms.templates.query({
             serviceName: this.$stateParams.serviceName
         }).$promise.then((templates) => {
             this.templates.raw = templates;
+            for (let model of this.templates.raw) {
+                this.getDetails(model).then((template) => {
+                    this.templates.paginated.push(template);
```

Comment:
You should not have to deal with pagination while using oui-datagrid...
You should pass `templates.raw` to datagrid as `row` attribute. And then, for each line loading, you can use `row-loader`: a function that you define in the controller, taking `$row` as parameter and returns a complete line (with all columns) wrapped in a promise. In so doing, `getDetails(model)` is only run for displayed rows.